security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -321,13 +336,69 @@ func (sc *SecretController) scrtDeleted(obj interface{}) {
 	saName := scrt.Annotations[ServiceAccountNameAnnotationKey]
 	if sa, err := sc.core.ServiceAccounts(scrt.GetNamespace()).Get(saName, metav1.GetOptions{}); err == nil {
 		log.Infof("Re-creating deleted secret %s/%s.", scrt.GetNamespace(), GetSecretName(saName))
-		if sc.istioEnabledObject(sa.GetObjectMeta()) {
+		if sc.citadelManagedObject(sa.GetObjectMeta()) {
 			sc.upsertSecret(saName, scrt.GetNamespace())
 		}
 		sc.monitoring.SecretDeletion.Inc()
 	}
 }
 
+func (sc *SecretController) namespaceUpdated(oldObj, newObj interface{}) {
+	oldNs := oldObj.(*v1.Namespace)
+	newNs := newObj.(*v1.Namespace)
+
+	oldManaged := sc.namespaceIsManaged(oldNs)
+	newManaged := sc.namespaceIsManaged(newNs)
+
+	if !oldManaged && newManaged {
+		sc.enableNamespaceRetroactive(newNs.GetName())
+		return
+	}
```

Comment:
Have we covered the managed --> unmanaged case?

Reply:
We decided that we will not perform retroactive secret deletion in the case that the namespace moves from `unmanaged` to `managed`, so there doesn't need to be a case to handle it (I'll write a comment to that effect)